Context.elicit overload chain is broken for mypy: bare string literals between the @overload stubs

## Summary

`Context.elicit`'s `@overload` chain is broken for mypy, because bare string literals sit *between* the overload stubs instead of inside their bodies. mypy stops collecting overloads at the first stray statement, so it only ever sees the **first** stub. Every other documented `response_type` is then rejected at the call site.

pyright and ty tolerate the interleaved statements and see all six overloads, so this shows up as a one-sided type-checker divergence that is easy to miss in CI.

## The shape

In `fastmcp_slim/fastmcp/server/context.py` on `main` (lines 957-1024), each stub's body is `...`, and the explanatory prose follows *after* the body, at class-body level:

```python
    @overload
    async def elicit(
        self,
        message: str,
        response_type: type[T],
        *,
        response_title: str | None = None,
        response_description: str | None = None,
    ) -> AcceptedElicitation[T] | DeclinedElicitation | CancelledElicitation: ...

    """The accepted elicitation will contain the response data"""   # <-- line 967, breaks the chain

    @overload
    async def elicit(
        self,
        message: str,
        response_type: list[str],
        ...
```

Five such literals, at lines 967, 979, 992, 1007, and 1022.

They are not docstrings. A docstring has to be the *first statement inside* the function body. These parse as standalone expression statements in the class body, between one overload and the next.

## Reproduction (no fastmcp needed)

```python
from typing import overload


@overload
def broken(x: None) -> int: ...


"""explanatory text that is NOT inside the overload body"""


@overload
def broken(x: list[str]) -> str: ...


def broken(x: None | list[str] = None) -> int | str:
    return 0 if x is None else "s"


broken(["a"])
```

```
$ mypy --strict repro.py
repro.py:5: error: Single overload definition, multiple required  [misc]
repro.py:12: error: Name "broken" already defined on line 5  [no-redef]
repro.py:12: error: Single overload definition, multiple required  [misc]
repro.py:19: error: Argument 1 to "broken" has incompatible type "list[str]"; expected "None"  [arg-type]

$ pyright repro.py
0 errors, 0 warnings, 0 informations
```

Moving each literal inside its stub body (docstring position, before the `...`) makes mypy report `Success: no issues found`.

## Real-world impact

Downstream code making a documented, correct call:

```python
CONFIRM_CHOICES: Final[list[str]] = ["CANCEL", "CONFIRM"]
result = await ctx.elicit(message, response_type=CONFIRM_CHOICES)
```

```
error: Argument "response_type" to "elicit" of "Context" has incompatible type
       "list[str]"; expected "None"  [arg-type]
```

There is no call-site workaround. We tested five formulations (inline list literal, non-`Final` module global, positional argument, a local `list()` copy, and a `StrEnum` passed as `type[T]`). All five fail under mypy and all five pass under pyright, because with only the first stub visible, every value that is not the first stub's type is rejected.

On the 3.x line that first stub is `response_type: None`, which is the form the docstring itself marks as deprecated:

> Passing `response_type=None` (or omitting it) is deprecated and will be removed in a future version.

So mypy users are currently pushed toward the one call shape the library is trying to retire.

## Affected versions

Verified by installing each into a clean directory and inspecting the shipped `context.py`:

| Version | Stray literals | Notes |
|---|---|---|
| `fastmcp-slim` 3.4.4 | yes | |
| `fastmcp-slim` 3.4.5 | 6 | |
| `fastmcp-slim` 3.4.6 (latest stable) | 6 | |
| `fastmcp-slim` 4.0.0b1 | 5 | `response_type: None` stub dropped, so the chain now starts at `type[T]` and the error becomes `expected "type[Never]"`. Still broken. |
| `main` @ HEAD | 5 | lines 967, 979, 992, 1007, 1022 |

## Suggested fix

Move each literal inside its stub, before the `...`:

```python
    @overload
    async def elicit(
        self,
        message: str,
        response_type: type[T],
        *,
        response_title: str | None = None,
        response_description: str | None = None,
    ) -> AcceptedElicitation[T] | DeclinedElicitation | CancelledElicitation:
        """The accepted elicitation will contain the response data"""
        ...
```

That keeps the prose exactly where a reader expects it, attaches it to the stub it describes, and restores the overload chain for mypy. No runtime or API change.

Happy to open a PR if that is useful.

--- a/fastmcp_slim/fastmcp/server/context.py
+++ b/fastmcp_slim/fastmcp/server/context.py
@@ -962,9 +962,8 @@ class Context:
         *,
         response_title: str | None = None,
         response_description: str | None = None,
-    ) -> AcceptedElicitation[T] | DeclinedElicitation | CancelledElicitation: ...
-
-    """The accepted elicitation will contain the response data"""
+    ) -> AcceptedElicitation[T] | DeclinedElicitation | CancelledElicitation:
+        """The accepted elicitation will contain the response data"""
 
     @overload
     async def elicit(
@@ -974,10 +973,9 @@ class Context:
         *,
         response_title: str | None = None,
         response_description: str | None = None,
-    ) -> AcceptedElicitation[str] | DeclinedElicitation | CancelledElicitation: ...
-
-    """When response_type is a list of strings, the accepted elicitation will
-    contain the selected string response"""
+    ) -> AcceptedElicitation[str] | DeclinedElicitation | CancelledElicitation:
+        """When response_type is a list of strings, the accepted elicitation will
+        contain the selected string response"""
 
     @overload
     async def elicit(
@@ -987,10 +985,9 @@ class Context:
         *,
         response_title: str | None = None,
         response_description: str | None = None,
-    ) -> AcceptedElicitation[str] | DeclinedElicitation | CancelledElicitation: ...
-
-    """When response_type is a dict mapping keys to title dicts, the accepted
-    elicitation will contain the selected key"""
+    ) -> AcceptedElicitation[str] | DeclinedElicitation | CancelledElicitation:
+        """When response_type is a dict mapping keys to title dicts, the accepted
+        elicitation will contain the selected key"""
 
     @overload
     async def elicit(
@@ -1000,12 +997,9 @@ class Context:
         *,
         response_title: str | None = None,
         response_description: str | None = None,
-    ) -> (
-        AcceptedElicitation[list[str]] | DeclinedElicitation | CancelledElicitation
-    ): ...
-
-    """When response_type is a list containing a list of strings (multi-select),
-    the accepted elicitation will contain a list of selected strings"""
+    ) -> AcceptedElicitation[list[str]] | DeclinedElicitation | CancelledElicitation:
+        """When response_type is a list containing a list of strings (multi-select),
+        the accepted elicitation will contain a list of selected strings"""
 
     @overload
     async def elicit(
@@ -1015,13 +1009,10 @@ class Context:
         *,
         response_title: str | None = None,
         response_description: str | None = None,
-    ) -> (
-        AcceptedElicitation[list[str]] | DeclinedElicitation | CancelledElicitation
-    ): ...
-
-    """When response_type is a list containing a dict mapping keys to title dicts
-    (multi-select with titles), the accepted elicitation will contain a list of
-    selected keys"""
+    ) -> AcceptedElicitation[list[str]] | DeclinedElicitation | CancelledElicitation:
+        """When response_type is a list containing a dict mapping keys to title dicts
+        (multi-select with titles), the accepted elicitation will contain a list of
+        selected keys"""
 
     async def elicit(
         self,

